src/API.jl

```diff
@@ -18,7 +19,7 @@ $(TYPEDSIGNATURES)
 Provides a suggestion based on random search to generate hyperparameter values.
 Also can generate trials to be evaluated from Dict of points.
 """
-function ask(space::Dict{Symbol, T} where T)::Trials.Trial
+function ask(space)::Trials.Trial
```

Comment:
why you didn't put type in here

Reply:
This is to support suggestions. Previously when we only accepted dicts then the suggestions would be dicts too.
Now, we support a dict or array of `AbstractDelayed` or even just an `AbstractDelayed` < the last one can correspond with getting hyperparams as a scalar, e.g. by passsing `HP.Uniform(:bla, 0., 1.)` `ask` will give us  values between 0 and 1:

```
julia> TreeParzen.ask(TreeParzen.HP.Uniform(:bla, 0., 1.)).hyperparams
0.04432218813798294

julia> TreeParzen.ask(TreeParzen.HP.Uniform(:bla, 0., 1.)).hyperparams
0.7229058081423172

julia> TreeParzen.ask(TreeParzen.HP.Uniform(:bla, 0., 1.)).hyperparams
0.8128138585478044

julia> 
```

what is the suggestion equivalent?

```
julia> TreeParzen.ask(1).hyperparams
1

julia> 

```

This can only work if the input is untyped

Reply:
(refer to the bit where the Trial hyperparams struct field was also changed to type `Any`)

Reply:
thanks for explaining